cpp/src/community/k_truss_impl.cuh

```diff
@@ -0,0 +1,929 @@
+                                 });
+
+      /*
+      //FIXME: Getting different results than the above
+      auto edges_with_triangle_last =
+        thrust::remove_if(handle.get_thrust_policy(),
+                          transposed_edge_triangle_count_pair_first,
+                          transposed_edge_triangle_count_pair_first + num_triangles.size(),
+                          [] __device__(auto e) {
+                            auto num_triangles = thrust::get<1>(e);
+                            return num_triangles == 0;
+                          });
+      */
+
+      num_edges_with_triangles = static_cast<size_t>(
+        thrust::distance(transposed_edge_triangle_count_pair_first, edges_with_triangle_last));
+
+      cugraph::edge_property_t<decltype(cur_graph_view), bool> edge_mask(handle, cur_graph_view);
+      // Set edge property to 'True' for all edges then mask out invalid edges which can be
+      // significantly smaller than the valid edges
+      cugraph::fill_edge_property(handle, cur_graph_view, true, edge_mask);
+
+      thrust::sort(handle.get_thrust_policy(),
+                   thrust::make_zip_iterator(edgelist_srcs.begin() + num_edges_with_triangles,
+                                             edgelist_dsts.begin() + num_edges_with_triangles),
+                   thrust::make_zip_iterator(edgelist_srcs.end(), edgelist_dsts.end()));
+
+      cugraph::edge_bucket_t<vertex_t, void, true, multi_gpu, true> edges_with_no_triangle(handle);
+      edges_with_no_triangle.insert(edgelist_srcs.begin() + num_edges_with_triangles,
+                                    edgelist_srcs.end(),
+                                    edgelist_dsts.begin() + num_edges_with_triangles);
+
+      // FIXME: Cannot modify an edgemask that is still attached.
+      // This can lead to race conditions
+      cugraph::transform_e(
+        handle,
+        cur_graph_view,
+        edges_with_no_triangle,
+        cugraph::edge_src_dummy_property_t{}.view(),
+        cugraph::edge_dst_dummy_property_t{}.view(),
+        cugraph::edge_dummy_property_t{}.view(),
+        [] __device__(auto src, auto dst, thrust::nullopt_t, thrust::nullopt_t, thrust::nullopt_t) {
+          return false;
+        },
+        edge_mask.mutable_view(),
+        false);  // FIXME: Remove expensive check. This is only here for debugging purposes
+                 // **********************************
+
+      cur_graph_view.attach_edge_mask(edge_mask.view());
+
+      edgelist_srcs.resize(num_edges_with_triangles, handle.get_stream());
+      edgelist_dsts.resize(num_edges_with_triangles, handle.get_stream());
+      num_triangles.resize(num_edges_with_triangles, handle.get_stream());
+
+      thrust::sort_by_key(handle.get_thrust_policy(),
+                          transposed_edge_first,
+                          transposed_edge_first + edgelist_srcs.size(),
+                          num_triangles.begin());
```

Comment:
That is correct. I realized that once we sort it, `stable_partition` and `remove_if` will preserve the sorting. But I just found an edge case. `nbr_intersection` requires the edgelist to be sorted by `src` otherwise returns an error. Therefore, I need to temporarily destroy the sorting of the invalid edge partition and once done with `nbr_intersection` sort it back by `dst`. Unless you have a better preference